src/test/addrman_tests.cpp

```diff
@@ -398,8 +405,8 @@ BOOST_AUTO_TEST_CASE(addrman_getaddr)
     // Test 25: Ensure GetAddr still returns 23% when addrman has many addrs.
     for (unsigned int i = 1; i < (8 * 256); i++) {
         int octet1 = i % 256;
```

Comment:
@jimmysong It just makes it less likely we will have a collision when adding addresses.

Changing
```cpp
int octet3 = (i / (256 * 2)) % 256;
```
to 
```cpp
int octet3 = (i / (256 * 256)) % 256;
```
would cause all of these generated addresses to map to the same source group in the new table as they would all have the same \16. 

Current behavior is intended to lower the chance that addresses are evicted when adding them by spreading them across more than one source group. A byproduct of this is also a more realistic test since the addresses returned by GetAddr() are now being drawn from several source groups.